nixos/modules/services/monitoring/watchdogd.nix

```diff
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.watchdogd;
+
+  mkPluginOpts = plugin: {
+    enable = mkEnableOption "watchdogd plugin ${plugin}";
+    interval = mkOption {
+      type = types.ints.unsigned;
+      default = 300;
+      description = ''
+        Amount of seconds between every poll.
+      '';
+    };
+    logmark = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to log current stats every poll interval.
+      '';
+    };
+    warning = mkOption {
+      type = types.ints.unsigned;
+      description = ''
+        The high watermark level. Alert sent to log.
+      '';
+    };
+    critical = mkOption {
+      type = types.ints.unsigned;
+      description = ''
+        The critical watermark level. Alert sent to log, followed by reboot or script action.
+      '';
+    };
+  };
+in {
+  options.services.watchdogd = {
+    enable = mkEnableOption "Advanced system & process supervisor";
+    package = mkPackageOptionMD pkgs "watchdogd" { };
+
+    timeout = mkOption {
+      type = types.ints.unsigned;
+      default = 15;
+      description = ''
+        The WDT timeout before reset.
+      '';
+    };
+    interval = mkOption {
+      type = types.ints.unsigned;
+      default = 5;
+      description = ''
+        The kick interval, i.e. how often {manpage}`watchdogd(8)` should reset the WDT timer.
+      '';
+    };
+
+    safeExit = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        With {var}`safeExit` enabled, the daemon will ask the driver to disable the WDT before exiting.
+        However, some WDT drivers (or hardware) may not support this.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
```

Comment:
While this is definitely more simple, I think it would be more semantically correct to set it to `nullOr lines`, set the default to `null` and use `optionalString` in `config`. Up to you.